packages/jsii-pacmak/lib/targets/golang/types/interface.ts

```diff
@@ -3,61 +3,121 @@ import { InterfaceType, Method, Property } from 'jsii-reflect';
 import { GoType } from './go-type';
 import { GoTypeRef } from './go-type-reference';
 import { Package } from '../package';
+import { TypeField } from './type-field';
+import { getFieldDependencies } from '../util';
+
+class InterfaceProperty implements TypeField {
+  public readonly name: string;
+  public readonly references?: GoTypeRef;
+
+  public constructor(
+    public readonly parent: Interface,
+    private readonly property: Property,
+  ) {
+    this.name = property.name;
+
+    if (property.type) {
+      this.references = new GoTypeRef(parent.parent.root, property.type);
+    }
+  }
+
+  public emit(code: CodeMaker) {
+    const propName = code.toPascalCase(this.name);
+    const type = new GoTypeRef(
+      this.parent.parent.root,
+      this.property.type,
+    ).scopedName(this.parent.parent);
+
+    code.line(`Get${propName}() ${type}`);
```

Comment:
For datatype interfaces, this should already be handled by https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/golang/types/go-type.ts#L64-L66

Reply:
Interface doesn't currently inherit from from struct. We can/should do that for property shared property emit logic but I'd say thats another PR.